src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java

```diff
@@ -54,4 +54,24 @@ public InnReachTransactionsDTO getAllTransactions(Integer offset, Integer limit,
     return transactionMapper.toDTOCollection(transactions);
   }
 
+  @Override
+  @Transactional
+  public InnReachTransactionDTO updateInnReachTransaction(UUID transactionId, InnReachTransactionDTO transaction) {
+    var oldTransaction = getInnReachTransaction(transactionId);
+
+    var oldTrackingId = oldTransaction.getTrackingId();
+    var oldCode = oldTransaction.getCentralServerCode();
+    var oldType = oldTransaction.getType().name();
+
+    var newTrackingId = transaction.getTrackingId();
+    var newCode = transaction.getCentralServerCode();
+    var newType = transaction.getType().name();
+
+    if (oldTrackingId.equals(newTrackingId) && oldCode.equals(newCode) && oldType.equals(newType)) {
+      repository.save(transactionMapper.toEntity(transaction));
+      return transaction;
+    }
```

Comment:
instead of addin a condition (which is not part of requirements) simply ignore new fields and use original ones. Also, I think it is more safe to update fields of existing entity (i.e. `oldTransaction`) instead of trying to save a new one

Reply:
implemented